### PR TITLE
Hash from string

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/1098)

### CODE REVIEW

File: model/src/main/kotlin/HashAlgorithm.kt [hunk 1]
```diff
@@ -24,7 +24,6 @@ import com.fasterxml.jackson.annotation.JsonValue
 
 enum class HashAlgorithm(private vararg val aliases: String) {
     UNKNOWN("", "UNKNOWN"),
-    MD2("MD2"),
```
Comment (mnonnenmacher) [inline model/src/main/kotlin/HashAlgorithm.kt]: Commit message nit: "I" -> "It"

File: analyzer/src/main/kotlin/managers/NPM.kt [hunk 2]
```diff
@@ -263,7 +263,7 @@ open class NPM(analyzerConfig: AnalyzerConfiguration, repoConfig: RepositoryConf
                                     }
 
                                     hash = dist["shasum"].textValueOrEmpty()
-                                    hashAlgorithm = HashAlgorithm.SHA1
+                                    hashAlgorithm = HashAlgorithm.fromHash(hash)
```
Comment (mnonnenmacher) [inline analyzer/src/main/kotlin/managers/NPM.kt]: The Bundler implementation also uses a hardcoded algorithm, should the same change be done there, too?

Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/NPM.kt]: Thanks for catching, done.

**Comment by sschuberth** [comment]
Merging despite the unrelated Travis CI failure.

**Pull request merged**